src/reducers/entities/posts.js

```diff
@@ -90,6 +90,35 @@ function handleReceivedPosts(posts = {}, postsInChannel = {}, action) {
     return {posts: nextPosts, postsInChannel: nextPostsForChannel};
 }
 
+function handlePostsFromSearch(posts = {}, postsInChannel = {}, action) {
+    const newPosts = action.data.posts;
+    let info = {posts, postsInChannel};
+    const postsForChannel = new Map();
+
+    for (const id in newPosts) {
+        if (newPosts.hasOwnProperty(id)) {
+            const nextPost = newPosts[id];
+            if (nextPost) {
+                const channelId = nextPost.channel_id;
+                if (postsForChannel.has(channelId)) {
+                    postsForChannel.set(channelId,
```

Comment:
Maybe change this to `postsForChannel.get(channelId)[id] = nextPost`? We don't need to worry about mutating the contents of `postsForChannel` since they're just used within this function